src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Transaction;
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void process(Transaction transaction) {
+        if (Storage.storage.get(transaction.getFruit()) == null) {

```

Comment:
You can use initialQuantity instead of Storage.storage.get(transaction.getFruit())